kernel/kernel-defaults/src/main/java/io/delta/kernel/defaults/internal/parquet/ParquetFileReader.java

```diff
@@ -16,77 +16,76 @@
+
 public class ParquetFileReader {
     private final Configuration configuration;
     private final int maxBatchSize;
 
     public ParquetFileReader(Configuration configuration) {
         this.configuration = requireNonNull(configuration, "configuration is null");
         this.maxBatchSize =
-            configuration.getInt("delta.kernel.default.parquet.reader.batch-size", 1024);
+                configuration.getInt("delta.kernel.default.parquet.reader.batch-size", 1024);
         checkArgument(maxBatchSize > 0, "invalid Parquet reader batch size: " + maxBatchSize);
     }
 
-    public CloseableIterator<ColumnarBatch> read(String path, StructType schema) {
-        BatchReadSupport batchReadSupport = new BatchReadSupport(maxBatchSize, schema);
-        ParquetRecordReader<Object> reader = new ParquetRecordReader<>(batchReadSupport);
+    public CloseableIterator<ColumnarBatch> read(
+            String path,
+            StructType schema,
+            Optional<Predicate> predicate) {
+
         final boolean hasRowIndexCol =
-            schema.indexOf(StructField.METADATA_ROW_INDEX_COLUMN_NAME) >= 0 &&
-            schema.get(StructField.METADATA_ROW_INDEX_COLUMN_NAME).isMetadataColumn();
-
-        Path filePath = new Path(URI.create(path));
-        try {
-            FileSystem fs = filePath.getFileSystem(configuration);
-            FileStatus fileStatus = fs.getFileStatus(filePath);
-            FileSplit fileSplit = new FileSplit(filePath, 0, fileStatus.getLen(), new String[0]);
-            reader.initialize(fileSplit, configuration, Reporter.NULL);
-        } catch (IOException | InterruptedException e) {
-            throw new RuntimeException(e);
-        }
+                schema.indexOf(StructField.METADATA_ROW_INDEX_COLUMN_NAME) >= 0 &&
+                        schema.get(StructField.METADATA_ROW_INDEX_COLUMN_NAME).isMetadataColumn();
 
         return new CloseableIterator<ColumnarBatch>() {
+            private final BatchReadSupport readSupport = new BatchReadSupport(maxBatchSize, schema);
+            private ParquetRecordReaderWrapper<Object> reader;
             private boolean hasNotConsumedNextElement;
 
             @Override
-            public void close()
-                throws IOException {
-                reader.close();
+            public void close() throws IOException {
+                Utils.closeCloseables(reader);
             }
 
             @Override
             public boolean hasNext() {
+                initParquetReaderIfRequired();
                 try {
                     if (hasNotConsumedNextElement) {
                         return true;
                     }
-                    hasNotConsumedNextElement = reader.nextKeyValue();
-                    return hasNotConsumedNextElement;
-                } catch (IOException | InterruptedException e) {
-                    throw new RuntimeException(e);
+                    return hasNotConsumedNextElement = reader.nextKeyValue() &&
```

Comment:
```java
return hasNotConsumedNextElement = reader.nextKeyValue() && reader.getCurrentValue() != null;
```

Do you _have_ to do the variable assignment within this predicate? Can you move it to its own line?

For example, I can't tell if it's supposed to be `==` instead of `=`